docs: Showcase temporal flamegraphs

mgmacias95 opened the pull request:
*Issue number of the reported bug or feature request: #392*

**Describe your changes**
Add a sample program that showcases temporal flamegraphs following indications at https://github.com/bloomberg/memray/issues/392#issuecomment-1592112729

**Testing performed**
Built docs and manually checked on my browser.

godlygeek commented:
Would you mind formatting the Python code with Black, and then adding a [DCO sign-off](https://github.com/bloomberg/memray/blob/main/CONTRIBUTING.md#contribution-licensing) to your commit message? You can add the DCO sign-off with `git commit -a --amend -s`

mgmacias95 commented:
hi @godlygeek,

I fixed the signed-off commit message, formatted the code using black, fixed linter issues and, in addition, I linked the example in the temporal flamegraph section in the docs. PTAL.

godlygeek reviewed (approved):
LGTM - thanks for the contribution, @mgmacias95!
